utils/configuration_test.go

```diff
@@ -485,3 +485,74 @@ func TestParseViperWithValidFile(t *testing.T) {
 
 	require.Equal(t, "debug", v.GetString("logging.level"))
 }
+
+func TestAdjustLogLevel(t *testing.T) {
```

Comment:
I'll let this slide in this review, but you need to decompose this into a data-provider like model. You are repeating yourself all over the place in this function.